internal/workflow/preprocessing.go

```diff
@@ -384,3 +315,95 @@ func systemError(
 	r.Outcome = OutcomeSystemError
 	return r
 }
+
+func writePREMISfile(ctx temporalsdk_workflow.Context, root string, sip sip.SIP) error {
```

Comment:
Could this function possibly not have the "root" argument and instead use `sip.Path` internally? Or would their values possibly be different when this function gets called?

Reply:
Oh yeah, that works. :+1: